projectq/ops/_command.py

```diff
@@ -243,7 +245,7 @@ def add_control_qubits(self, qubits):
         self._control_qubits.extend(
             [WeakQubitRef(qubit.engine, qubit.id) for qubit in qubits])
         self._control_qubits = sorted(self._control_qubits, key=lambda x: x.id)
-
+        self.ctrl_state = ctrl_state
```

Comment:
Here you need to add some additional checks to make sure that two `Control` block do not have contradicting state for the relevant qubits

Reply:
Plus this code is currently **wrong**: you need to also extend the `self.ctrl_state` if you add a new control qubits and not simply replace the `self.ctrl_state` with the new control state.